dialog/base.go

```diff
@@ -37,19 +38,33 @@ type dialog struct {
 	dismiss        *widget.Button
 
 	response  chan bool
+	mu        sync.Mutex
 	responded bool
 	parent    fyne.Window
 }
 
 func (d *dialog) wait() {
-	select {
```

Comment:
AFAICS the changes regarding the goroutine leak are independent from the change to the `ShowCustom*` functions.
Please separate different topics into different commits in future contributions, which we greatly appreciate.